### PR TITLE
[Storage] `az storage copy`: Add account credentials support and deprecate `--source-local-path`, `--destination-local-path`, `--destination-account-name`

### PR DESCRIPTION
**Pull request by Juliehzl**
**Description**  

dupe of https://github.com/Azure/azure-cli/pull/14636
- refactor the code
- will deprecate `--source-local-path` , redirect to `--source`
- will deprecate `--destination-local-path`, redirect to `--destination`
- will deprecate `--destination-account-name`, redirect to `--account-name`
- Add account credential related argument

**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S175

File: src/azure-cli/azure/cli/command_modules/storage/_help.py [hunk 1]
```diff
@@ -1117,15 +1117,15 @@
   - name: Upload an entire directory to Azure Blob using url.
     text: az storage copy -s /path/to/dir -d https://[account].blob.core.windows.net/[container]/[path/to/directory] --recursive
   - name: Upload an entire directory to Azure Blob using account name and container name.
-    text: az storage copy --source-local-path /path/to/dir --destination-account-name mystorageaccount --destination-container mycontainer --recursive
+    text: az storage copy -s /path/to/dir --destination-account-name mystorageaccount --destination-container mycontainer --recursive
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/storage/_help.py:1120]: There are two places that haven't been changed. Are they in line with expectations?
https://github.com/Azure/azure-cli/blob/cfa31a7439e3d7dd4ff38fe03afd8a753a9cb19b/src/azure-cli/azure/cli/command_modules/storage/_help.py#L1158
https://github.com/Azure/azure-cli/blob/cfa31a7439e3d7dd4ff38fe03afd8a753a9cb19b/src/azure-cli/azure/cli/command_modules/storage/_help.py#L1114

File: src/azure-cli/azure/cli/command_modules/storage/_validators.py [hunk 2]
```diff
@@ -1342,3 +1342,93 @@ def validate_or_policy(namespace):
 
         if "policyId" in or_policy.keys() and or_policy["policyId"]:
             namespace.policy_id = or_policy['policyId']
+
+
+def get_url_with_sas(cmd, namespace, url=None, container=None, blob=None, share=None, file_path=None):
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/storage/_validators.py:1347]: The container and share parameters cannot be passed in at the same time, right? Could we add a check for this?

Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/storage/_validators.py:1347]: Can this method be reused in other places? If so, maybe we can consider putting it into the utils~

File: src/azure-cli/azure/cli/command_modules/storage/operations/azcopy.py [hunk 3]
```diff
@@ -11,82 +11,75 @@
...
+    if url is not None:
+        if "?" in url:  # sas token exists
+            return url
+        # validate source is uri or local path
+        storage_pattern = re.compile(r'https://(.*?)\.(blob|dfs|file).%s' % storage_endpoint)
+        result = re.findall(storage_pattern, url)
+        if result:  # source is URL
+            storage_info = result[0]
+            account_name = storage_info[0]
+            if storage_info[1] in ['blob', 'dfs']:
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/storage/operations/azcopy.py:39]: Could we consider reuse some common logics from https://github.com/Azure/azure-cli/blob/cfa31a7439e3d7dd4ff38fe03afd8a753a9cb19b/src/azure-cli/azure/cli/command_modules/storage/_validators.py#L1347

File: src/azure-cli/azure/cli/command_modules/storage/_help.py (hunk 1 above)
Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/storage/_help.py:1120]: done

File: src/azure-cli/azure/cli/command_modules/storage/_validators.py (hunk 2 above)
Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/storage/_validators.py:1347]: sure

**Review by zhoxing-ms — APPROVED** [review]
LGTM

**Review by jsntcy — APPROVED** [review]
:shipit:

**Pull request merged**